internal/juju/client.go

```diff
@@ -182,7 +185,16 @@ func (sc *sharedClient) GetConnection(modelName *string) (api.Connection, error)
 	return conn, nil
 }
 
-func (sc *sharedClient) ModelUUID(modelName string) (string, error) {
+// ModelUUID returns the model uuid for the provided modelIdentifier.
+// The modelIdentifier can be a model name or model uuid. If a name
+// is provided, first search the modelUUIDCache for the uuid. If it's
+// not found, fill the model cache and try again. If the modelIdentifier
+// is a uuid, return that without verification.
+func (sc *sharedClient) ModelUUID(modelIdentifier string) (string, error) {
```

Comment:
suggestion:
I think we should remove the duplication from ModelUUID, ModelType, and ModelName. They all effectively,
1. Check if the modelIdentifier is a UUID/name
2. Check modelUUID cache.

Confusingly, only the ModelUUID and ModelName functions refill the cache if the name was not found. If possible, we could refactor this to a separate function to make it less prone to error when we make changes.

Reply:
I do not see a duplication of the data. All three values are specific to one model, modulo owner-name. Not all resources have the model UUID at this time. I had to make the scope of this change smaller to better understand how to handle upgrade and import cases for existing plans/state.

I can add some error checking, though it "should" be impossible for a model to be added which does not contain all three values, as they are added either after a creating a model via TF, or by ListModels from juju. 

The first thing the ModelUUID method does is return the UUID if the Identifier is a UUID.